test/test_integration.rb

```diff
@@ -374,4 +374,30 @@ def test_not_accepts_new_connections_after_term_signal
     Process.wait(@server.pid)
     @server = nil # prevent `#teardown` from killing already killed server
   end
+
+  def test_no_zombie_children
+    skip_on :jruby, :windows
+
+    worker_pids = []
+    server = server("-w 2 test/rackup/hello.ru")
+    # Get the PIDs of the child workers.
+    while worker_pids.size < 2
+      next unless line = server.gets.match(/pid: (\d+)/)
+      worker_pids << line.captures.first.to_i
+    end
+    # Signal the workers to terminate, and wait for them to die.
+    worker_pids.each { |pid| Process.kill :TERM, pid }
+    sleep 1
+    # Check if the process remains in the process table.
```

Comment:
This test implies that on current master, process.kill :term creates a zombie. Is that really true?

Reply:
Ok, it is true. Seen as much with your issue description + repro'd myself. Ugh, that's awful.